tests/integration/extenders/ModelTest.php

```diff
@@ -0,0 +1,205 @@
+            return $user->hasMany(Discussion::class, 'user_id');
+        }));
+
+        $this->prepDB();
+
+        $user = User::find(1);
+
+        $this->assertEquals([], $user->customRelation()->get()->toArray());
+    }
+
+    /**
+     * @test
+     */
+    public function custom_relationship_exists_and_can_return_instances_if_added()
+    {
+        $this->extend((new Extend\Model(User::class))->relationship('customRelation', function (User $user) {
+            return $user->hasMany(Discussion::class, 'user_id');
+        }));
+
+        $this->prepDB();
+
+        $this->prepareDatabase([
+            'discussions' => [
+                ['id' => 1, 'title' => __CLASS__, 'created_at' => Carbon::now()->toDateTimeString(), 'user_id' => 1, 'first_post_id' => 1, 'comment_count' => 1]
+            ]
+        ]);
+
+        $user = User::find(1);
+
+        $this->assertNotEquals([], $user->customRelation()->get()->toArray());
+        $this->assertContains(json_encode(__CLASS__), json_encode($user->customRelation()->get()));
+    }
+
+    /**
+     * @test
+     */
+    public function custom_relationship_does_not_exist_if_added_to_unrelated_model()
+    {
+        $this->extend((new Extend\Model(User::class))->relationship('customRelation', function (User $user) {
+            return $user->hasMany(Discussion::class, 'user_id');
+        }));
+
+        $this->prepDB();
+        $this->prepareDatabase([
+            'groups' => [
+                $this->adminGroup()
+            ]
+        ]);
+
+        $group = Group::find(1);
+
+        $this->expectException(\BadMethodCallException::class);
+        $group->customRelation();
+    }
+
+    /**
+     * @test
+     */
+    public function custom_default_attribute_doesnt_exist_if_not_set()
+    {
+        $group = new Group;
+
+        $this->assertNotEquals('Custom Default', $group->name_singular);
+    }
+
+    /**
+     * @test
+     */
+    public function custom_default_attribute_works_if_set()
+    {
+        $this->extend((new Extend\Model(Group::class))->configureDefaultAttributes(function ($defaults) {
+            $defaults['name_singular'] = 'Custom Default';
+
+            return $defaults;
+        }));
+
+        $this->app();
+
+        $group = new Group;
+
+        $this->assertEquals('Custom Default', $group->name_singular);
+    }
+
+    /**
+     * @test
+     */
+    public function custom_default_attribute_doesnt_work_if_set_on_unrelated_model()
+    {
+        $this->extend((new Extend\Model(Group::class))->configureDefaultAttributes(function ($defaults) {
+            $defaults['name_singular'] = 'Custom Default';
+
+            return $defaults;
+        }));
+
+        $this->app();
+
+        $user = new User;
+
+        $this->assertNotEquals('Custom Default', $user->name_singular);
+    }
+
+    /**
+     * @test
+     */
+    public function custom_date_doesnt_exist_if_not_set()
+    {
+        $post = new Post;
```

Comment:
Please run `$this->app()` in this test (and some of the other ones) as well, for maximum comparability. :grin: